projects/plugins/jetpack/extensions/shared/components/product-management-controls/inspector-control.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * WordPress dependencies
+ */
+import { InspectorControls } from '@wordpress/block-editor';
+import {
+	Button,
+	PanelBody,
+	PanelRow,
+	SelectControl,
+	TextControl,
+	ExternalLink,
+	Placeholder,
+	Spinner,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { lock } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { API_STATE_NOT_REQUESTING, API_STATE_REQUESTING } from './constants';
+import { CURRENCY_OPTIONS } from '../../currencies';
+import { store as membershipProductsStore } from '../../../store/membership-products';
+
+export default function ProductManagementInspectorControl( {
+	allowCreateOneTimeInterval,
+	setSelectedProductId,
+} ) {
+	const siteSlug = useSelect( select => select( membershipProductsStore ).getSiteSlug() );
+	const { saveProduct } = useDispatch( membershipProductsStore );
+
+	const [ apiState, setApiState ] = useState( API_STATE_NOT_REQUESTING );
+	const [ title, setTitle ] = useState( __( 'Monthly Subscription', 'jetpack' ) );
+	const [ currency, setCurrency ] = useState( 'USD' );
+	const [ price, setPrice ] = useState( 5 );
+	const [ interval, setInterval ] = useState( '1 month' );
+
+	const intervalOptions = [
+		{ label: __( 'Month', 'jetpack' ), value: '1 month' },
+		{ label: __( 'Year', 'jetpack' ), value: '1 year' },
+	];
+	if ( allowCreateOneTimeInterval ) {
+		intervalOptions.push( { label: __( 'One-Time Payment', 'jetpack' ), value: 'one-time' } );
+	}
+
+	const handleSubmit = event => {
+		event.preventDefault();
+		setApiState( API_STATE_REQUESTING );
+		saveProduct( { title, currency, price, interval }, setSelectedProductId, success => {
+			setApiState( API_STATE_NOT_REQUESTING );
+			if ( success ) {
+				setPrice( 5 );
+				setTitle( '' );
+			}
+		} );
+	};
+
+	return (
+		<InspectorControls>
+			{ siteSlug && (
+				<ExternalLink
+					href={ `https://wordpress.com/earn/payments/${ siteSlug }` }
+					className={ 'product-management-control-inspector__link-to-earn' }
+				>
+					{ __( 'Manage your subscriptions.', 'jetpack' ) }
+				</ExternalLink>
+			) }
+			<PanelBody
+				title={ __( 'Add a new subscription', 'jetpack' ) }
```

Comment:
Maybe we could change this text for:
`Add this payment plan`
Not all the plans we can configure are subscriptions, it's strange to add a one-time payment and then click on "Add a new subscription".

Reply:
I have intentionally not done any copy changes, as that will require some specific handling that I'd rather do in a follow up.